core/mapserver/mapserver.go

```diff
@@ -70,12 +70,19 @@ func (m *mapServer) signRoot(ctx context.Context) (*trillian.SignedMapRoot, erro
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if retErr != nil {
+			if rbErr := txn.Rollback(); rbErr != nil {
+				retErr = fmt.Errorf("%v, Rollback(): %v", retErr, rbErr)
+			}
+		}
+	}()
 	root, err := m.tree.ReadRootAt(txn, epoch)
 	if err != nil {
-		return nil, fmt.Errorf("ReadRootAt err: %v", err)
+		return nil, fmt.Errorf("ReadRootAt(%v): %v", epoch, err)
```

Comment:
I prefer shorter error messages because they all get chained together:
`Dequeue(): CreateEpoch(): Commit(): table locked: SMH`
Seems cleaner than
`Dequeue() failed: CreateEpoch() failed: Commit() failed: table locked: SMH`
What do you think?